src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -204,6 +204,9 @@ public BasePanel(JabRefFrame frame, BibDatabaseContext bibDatabaseContext) {
         Objects.requireNonNull(bibDatabaseContext);
 
         this.bibDatabaseContext = bibDatabaseContext;
+        DatabaseChangeListener databaseChangeListener = new DatabaseChangeListener(this);
```

Comment:
Thanks, you are right! I started out with the intention of not making the BasePanel any bigger than necessary. That's why I added a listener class. But since this class turned out to be so dead simple, it can just as well go into BasePanel. 

I fixed this in the way you suggested.